File: frontend/src/app/models/auth.models.ts
// ─────────────────────────────────────────────────────────────────────────────
// auth.models.ts
// Shared TypeScript interfaces that describe the shape of data sent to and
// received from the backend API. Using interfaces keeps the rest of the code
// type-safe — TypeScript will warn you if you use a wrong field name or type.
// ─────────────────────────────────────────────────────────────────────────────

// Data sent to POST /api/auth/login
export interface LoginRequest {
  username: string;
  password: string;
}

// Data sent to POST /api/auth/register
export interface RegisterRequest {
  username: string;
  password: string;
}

// Data returned by the backend after a successful login or register.
// The token is a JWT string the frontend stores and sends with future requests.
export interface AuthResponse {
  token: string;
  username: string;
}

File: backend/Models/AuthModels.cs
// ─────────────────────────────────────────────────────────────────────────────
// AuthModels.cs
// Plain C# classes (DTOs — Data Transfer Objects) that represent the JSON
// bodies sent in HTTP requests and responses. Keeping them separate from the
// internal User model means the API contract is independent of the database model.
// ─────────────────────────────────────────────────────────────────────────────

using System.ComponentModel.DataAnnotations;

namespace AuthApi.Models
{
    // Body expected by POST /api/auth/login
    public class LoginRequest
    {
        [Required]  // Returns 400 Bad Request automatically if this field is missing.
        public string Username { get; set; }

        [Required]
        public string Password { get; set; }
    }

    // Body expected by POST /api/auth/register
    public class RegisterRequest
    {
        [Required]
        public string Username { get; set; }

        [Required]
        [MinLength(8)]  // Returns 400 if the password is shorter than 8 characters.
        public string Password { get; set; }
    }

    // JSON returned to the client after a successful login or register.
    public class AuthResponse
    {
        public string Token { get; set; }     // The JWT the client must store and reuse.
        public string Username { get; set; }  // Convenient for the UI to display the name.
    }
}
